index.js

```diff
@@ -222,7 +233,9 @@ module.exports = (function () {
                 return retry_loop();
               }
 
-              save_response_in_cache(options.method, options.url, options.auth.bearer, res, body);
+              if (res.statusCode >= 200 && res.statusCode < 300){
+                save_response_in_cache(options.method, options.url, options.auth.bearer, res, body);
```

Comment:
FWIW, I think this sounds brilliant.  We need at least one of caching for non-200s or exponential backoff on retries to avoid a DOS-on-failure cycle.